servicetalk-http-router-jersey/src/main/java/io/servicetalk/http/router/jersey/DefaultJerseyStreamingHttpRouter.java

```diff
@@ -251,6 +259,118 @@ private void handle0(final HttpServiceContext serviceCtx, final StreamingHttpReq
         applicationHandler.handle(containerRequest);
     }
 
+    /**
+     * {@link ContainerRequest#close()} may get called outside the thread that executes the
+     * {@link ApplicationHandler#handle(ContainerRequest)}. As a result, the close can be racy when the
+     * {@link org.glassfish.jersey.message.internal.InboundMessageContext} is accessed at the same time.
+     * This wrapper allows the {@link #close()} to be deferred after the reading is done by handing the {@link #close()}
+     * over to the {@link ApplicationHandler#handle(ContainerRequest)} owner thread. This also offers better thread
+     * visibility between the threads and the unsafely accessed variables.
+     */
+    private static final class CloseSignalHandoffAbleContainerRequest extends ContainerRequest {
+        private static final AtomicReferenceFieldUpdater<CloseSignalHandoffAbleContainerRequest, State> stateUpdater =
+                newUpdater(CloseSignalHandoffAbleContainerRequest.class, State.class, "state");
+
+        private enum State {
+            INIT,
+            READING,
+            PENDING_CLOSE,
+            CLOSED
+        }
+
+        private volatile State state = State.INIT;
+
+        private CloseSignalHandoffAbleContainerRequest(final URI baseUri, final URI requestUri, final String httpMethod,
+                                                      final SecurityContext securityContext,
+                                                      final PropertiesDelegate propertiesDelegate,
+                                                      @Nullable final Configuration configuration) {
+            super(baseUri, requestUri, httpMethod, securityContext, propertiesDelegate, configuration);
+        }
+
+        public <T> T readEntity(final Class<T> rawType) {
+            return readEntity(rawType, getPropertiesDelegate());
+        }
+
+        public <T> T readEntity(final Class<T> rawType, final Annotation[] annotations) {
+            return readEntity(rawType, annotations, getPropertiesDelegate());
+        }
+
+        public <T> T readEntity(final Class<T> rawType, final Type type) {
+            return readEntity(rawType, type, getPropertiesDelegate());
+        }
+
+        public <T> T readEntity(final Class<T> rawType, final Type type, final Annotation[] annotations) {
+            return readEntity(rawType, type, annotations, getPropertiesDelegate());
```

Comment:
The overrides unfortunately exist because the original implementation is invoking the method as `super.readEntity...`, therefore if I don't override the method of importance is never invoked from the other overloads.
Good suggestion about capturing the reason in a comment, will do.